src/main/java/org/jenkinsci/plugins/pagerduty/changeevents/ChangeEventBuilder.java

```diff
@@ -0,0 +1,101 @@
+package org.jenkinsci.plugins.pagerduty.changeevents;
+
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.AbstractProject;
+import hudson.model.AbstractBuild;
+import hudson.tasks.Builder;
+import hudson.model.BuildListener;
+import hudson.tasks.BuildStepDescriptor;
+import hudson.util.FormValidation;
+import jenkins.model.Jenkins;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.QueryParameter;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.annotation.Nonnull;
+
+/**
+ * A build step for recording a PagerDuty Change Event.
+ *
+ * This is intended to be used as a regular build step. It takes summary text
+ * and makes an API call to PagerDuty's Change Events API.
+ *
+ * See https://developer.pagerduty.com/docs/events-api-v2/send-change-events/
+ */
+
+public class ChangeEventBuilder extends Builder  {
+    /**
+     * The integration key that identifies the service the change occurred on.
+     */
+    private final String integrationKey;
+
+    /**
+     * custom event data that can be passed on to set as summary
+     */
+    private   String summaryText ;
+
+    @DataBoundConstructor
+    public ChangeEventBuilder(String integrationKey) {
+	this.integrationKey = integrationKey;
+    }
+
+    @Override
+    public boolean perform(AbstractBuild build, Launcher launcher, BuildListener listener) {
+	new ChangeEventSender().send(integrationKey, summaryText, build, listener);
+	return true;
+    }
+
+    public String getIntegrationKey() {
+	return integrationKey;
+    }
+
+    public String getSummaryText() {
+	return summaryText;
+    }
+
+    @DataBoundSetter
+    public void setSummaryText(String summaryText) {
+	this.summaryText = summaryText;
+    }
+
+    @Extension
+    public static final class DescriptorImpl extends BuildStepDescriptor<Builder> {
+	@Nonnull
+	public String getDisplayName() {
+	    return "PagerDuty Change Event Builder";
```

Comment:
I think you and I talked about this last week when we were pairing, so I may have made a mistake.  Seems to me that this should just be called "PagerDuty Change Event". Yes ,it IS a builder, but this displays on screen to the user.  No other steps are called "builder".  I could also be persuaded to go with something that makes it a little more clear what is going to happen, such as "Send PagerDuty Change Event".

Reply:
I missed that, I'll change it